daemon/libnetwork/network_windows.go

```diff
@@ -52,6 +50,9 @@ func (n *Network) startResolver() {
 	if n.networkType == "ics" {
 		return
 	}
+	if n.DriverOptions()[windows.DisableDNS] == "true" {
+		return
+	}
```

Comment:
I don't see how this relates to the IPAM improvement. Isn't that a separate bug fix? If so, could you open another PR and add a regression test please?